promgen/static/js/promgen.vue.js

```diff
@@ -27,72 +51,33 @@ const app = new Vue({
             let tgt = document.getElementById(target);
             tgt.classList.toggle('collapse');
         },
-        silenceExpire: function (id) {
-            fetch('/proxy/v1/silences/' + id, { method: 'DELETE' }).then(function (response) {
-                location.reload();
-            })
-        },
-        silenceChangeEvent: function (event) {
-            this.newSilence[event.target.name] = event.target.value;
-        },
-        silenceSubmit: function (event) {
-            let this_ = this;
-            fetch('/proxy/v1/silences', { method: 'POST', body: JSON.stringify(this.newSilence) })
-                .then(function (response) {
-                    if (response.ok) {
-                        location.reload();
-                    } else {
-                        return response.json()
-                    }
-                })
-                .then(function (result) {
-                    this_.globalMessages = [];
-                    for (key in result.messages) {
-                        this_.$set(this_.globalMessages, key, result.messages[key]);
-                    }
-                })
-        },
-        silenceRemoveLabel: function (label) {
-            console.debug('silenceRemoveLabel', label)
-            this.$delete(this.newSilence.labels, label)
-        },
-        showSilenceForm: function (event) {
-            this.$set(this.components, 'silence-form', true);
-            scroll(0, 0);
-        },
-        silenceAppendLabel: function (event) {
-            console.debug('silenceAppendLabel', event.target.dataset);
-            this.$set(this.newSilence.labels, event.target.dataset.label, event.target.dataset.value);
-            this.showSilenceForm(event);
+        expireSilence(id) {
+            fetch(`/proxy/v1/silences/${id}`, { method: 'DELETE' })
+                .then(() => location.reload());
         },
-        silenceSelectedHosts: function (event) {
-            this.$set(this.newSilence, 'labels', {});
-            this.$set(this.newSilence.labels, "instance", this.selectedHosts.join("|"));
-            for (key in event.target.dataset) {
-                this.$set(this.newSilence.labels, key, event.target.dataset[key]);
+        setSilenceLabels(labels) {
+            if (labels.target) {
+                silenceStore.setLabels(labels.target.dataset);
+            } else {
+                silenceStore.setLabels(labels);
             }
```

Comment:
Another option to do this would be just pass `$event.target.dataset` at the call sites. WDYT?

Reply:
Yeah, that's one of the few places I'm unsure about. Could also have two methods

```javascript
        setSilenceLabels(labels) {
            silenceStore.setLabels(labels);
            silenceStore.showForm();
            scroll(0, 0);
        },
        setSilenceDataset(ele) {
            this.setSilenceLabels(ele.target.dataset);
        },
```

That might be cleaner